admin_app/src/app/playground/components/PlaygroundComponents.tsx

```diff
@@ -227,8 +233,51 @@ const MessageSkeleton = () => {
   );
 };
 
-const MessageBox = (message: Message) => {
+const MessageBox = ({
+  message,
+  onFeedbackSend,
+}: {
+  message: Message;
+  onFeedbackSend: (
+    message: ResponseMessage,
+    feedbackSentiment: FeedbackSentimentType,
+  ) => void;
+}) => {
   const [open, setOpen] = useState(false);
+  const [thumbsUp, setThumbsUp] = useState(false);
+  const [thumbsDown, setThumbsDown] = useState(false);
+
+  const feedbackMapping = {
+    positive: {
+      state: thumbsUp,
+      setState: setThumbsUp,
+      onIcon: <ThumbUpAltIcon fontSize="small" />,
+      offIcon: <ThumbUpOffAltIcon fontSize="small" />,
+    },
+    negative: {
+      state: thumbsDown,
+      setState: setThumbsDown,
+      onIcon: <ThumbDownAltIcon fontSize="small" />,
+      offIcon: <ThumbDownOffAltIcon fontSize="small" />,
+    },
+  };
+
+  const handleFeedback = (feedbackType: FeedbackSentimentType) => {
+    const { state, setState } = feedbackMapping[feedbackType];
+
+    if (state) {
+      console.log(`Already sent ${feedbackType} feedback`);
+    } else {
+      setState(true);
+      return onFeedbackSend(message as ResponseMessage, feedbackType);
+    }
+  };
```

Comment:
@sidravi1 I have added these. what do you think? (all the console logs should be snack messages! But I have left it in back log :/)

Reply:
Looks good!